rotkehlchen/chain/evm/decoding/cowswap/decoder.py

```diff
@@ -369,7 +370,13 @@ def _decode_normal_claim(self, context: DecoderContext) -> DecodingOutput:
         raw_amount = hex_or_bytes_to_int(context.tx_log.data[128:160])
         amount = asset_normalized_value(amount=raw_amount, asset=self.vcow_token)
         airdrop_identifier: Literal['cow_mainnet', 'cow_gnosis'] = 'cow_mainnet' if self.evm_inquirer.chain_id == ChainID.ETHEREUM else 'cow_gnosis'  # noqa: E501
+        claim_has_payment = hex_or_bytes_to_int(context.tx_log.data[32:64]) in (1, 2)
         for event in context.decoded_events:
+            # Claim event always follows payment. Continue on payment, break on claim.
+            if claim_has_payment and event.event_type == HistoryEventType.SPEND and event.counterparty != CPT_GAS:  # noqa: E501
```

Comment:
I added it later but also requested to check against the location label since the address making the payment should be in the data too